airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/DefaultConfigPersistence.java

```diff
@@ -119,22 +230,24 @@ public DefaultConfigPersistence(final Path storageRoot, final JsonSchemaValidato
     }
   }
 
-  private <T> void writeConfigInternal(ConfigSchema configType, String configId, T config) throws JsonValidationException, IOException {
+  private <T> void writeConfigInternal(ConfigSchema configType, String configId, T config, Optional<Path> rootOverride)
```

Comment:
it is an anti-pattern to use Optional as argument. It is supposed to be used as a return type instead. You can just pass `null` and have an overload without the parameter, it makes the caller site cleaner.

Reply:
actually the better thing to do here is to simply make these methods to never use the classe member for the path and only use the `root` argument. Maybe they can even be `static` if path is the only thing they interact with.